Casting a seq to another seq generates invalid code with --newruntime

<!-- Think about the title, twice. -->

<!-- Summarize the problem here, keep it short and simple. -->
Casting a seq or a string to another seq generates invalid C code with --newruntime


### A couple of examples:
seq -> seq
<!-- Paste your example in the code-block below. -->
```nim
discard cast[seq[uint8]](@[1])
```
string -> seq
```nim
discard cast[seq[uint8]]("test")
```

### Current Output
```
Error: execution of an external compiler program 'gcc -c  -w  -I/home/clyybber/builds/nim/lib -I/home/clyybber/projects/tests -o /home/clyybber/.cache/nim/convrt_d/convrt.c.o /home/clyybber/.cache/nim/convrt_d/convrt.c' failed with exit code: 1

/home/clyybber/.cache/nim/convrt_d/convrt.c: In function ‘NimMainModule’:
/home/clyybber/.cache/nim/convrt_d/convrt.c:157:2: error: conversion to non-scalar type requested
  (void)(((tySequence_6H5Oh5UUvVCLiakt9aTwtUQ) (TM_KUSKOxFeGyNT8JCQmxu3cw_2)));
  ^
```


### Expected Output
<!-- What should be the correct output? -->
Compile successfully

### Additional Information
Casting an array to a seq does not generate invalid code:
```nim
discard cast[seq[uint8]]([1])
```

<!--- For Example:
* Your Nim version (output of `nim -v`).
* Was it working in the previous Nim releases?
* A link to a related issue or discussion.
-->
```
$ nim -v
Nim Compiler Version 0.19.9
```

--- a/compiler/ccgexprs.nim
+++ b/compiler/ccgexprs.nim
@@ -1839,6 +1839,9 @@ proc genSomeCast(p: BProc, e: PNode, d: var TLoc) =
     if srcTyp.kind in {tyPtr, tyPointer} and etyp.kind in IntegralTypes:
       putIntoDest(p, d, e, "(($1) (ptrdiff_t) ($2))" %
           [getTypeDesc(p.module, e.typ), rdCharLoc(a)], a.storage)
+    elif etyp.kind == tySequence:
+      putIntoDest(p, d, e, "(($1_Content*) ($2.p))" %
+          [getTypeDesc(p.module, e.typ), rdCharLoc(a)], a.storage)
     else:
       putIntoDest(p, d, e, "(($1) ($2))" %
           [getTypeDesc(p.module, e.typ), rdCharLoc(a)], a.storage)

--- a/compiler/ccgexprs.nim
+++ b/compiler/ccgexprs.nim
@@ -1839,7 +1839,7 @@ proc genSomeCast(p: BProc, e: PNode, d: var TLoc) =
     if srcTyp.kind in {tyPtr, tyPointer} and etyp.kind in IntegralTypes:
       putIntoDest(p, d, e, "(($1) (ptrdiff_t) ($2))" %
           [getTypeDesc(p.module, e.typ), rdCharLoc(a)], a.storage)
-    elif etyp.kind == tySequence:
+    elif p.config.selectedGc == gcDestructors and etyp.kind == tySequence:
       putIntoDest(p, d, e, "(($1_Content*) ($2.p))" %
           [getTypeDesc(p.module, e.typ), rdCharLoc(a)], a.storage)
     else:

--- a/compiler/ccgexprs.nim
+++ b/compiler/ccgexprs.nim
@@ -1840,7 +1840,7 @@ proc genSomeCast(p: BProc, e: PNode, d: var TLoc) =
       putIntoDest(p, d, e, "(($1) (ptrdiff_t) ($2))" %
           [getTypeDesc(p.module, e.typ), rdCharLoc(a)], a.storage)
     elif p.config.selectedGc == gcDestructors and etyp.kind == tySequence:
-      putIntoDest(p, d, e, "(($1_Content*) ($2.p))" %
+      putIntoDest(p, d, e, "(*($1*) (&$2))" %
           [getTypeDesc(p.module, e.typ), rdCharLoc(a)], a.storage)
     else:
       putIntoDest(p, d, e, "(($1) ($2))" %

--- a/tests/destructor/tv2_cast.nim
+++ b/tests/destructor/tv2_cast.nim
@@ -1,0 +1,8 @@
+discard """
+  cmd: '''nim c --newruntime $file'''
+"""
+
+# bug #11018
+discard cast[seq[uint8]](@[1])
+discard cast[seq[uint8]]("test")
+discard cast[seq[uint8]]([1])

--- a/tests/destructor/tv2_cast.nim
+++ b/tests/destructor/tv2_cast.nim
@@ -1,7 +1,7 @@
 discard """
   cmd: '''nim c --newruntime $file'''
   output: '''@[1]
-@[108, 111, 108]'''
+@[116, 101, 115, 116]'''
 """
 
 # bug #11018

--- a/tests/destructor/tv2_cast.nim
+++ b/tests/destructor/tv2_cast.nim
@@ -1,8 +1,11 @@
 discard """
   cmd: '''nim c --newruntime $file'''
+  output: '''@[1]
+@[108, 111, 108]'''
 """
 
 # bug #11018
 discard cast[seq[uint8]](@[1])
 discard cast[seq[uint8]]("test")
-discard cast[seq[uint8]]([1])
+echo cast[seq[uint8]](@[1])
+echo cast[seq[uint8]]("test")

